pkg/cleaner/cleaner.go

```diff
@@ -21,11 +21,19 @@ const (
 )
 
 type Ref struct {
-	mut *sync.Mutex
+	mut *sync.RWMutex
```

Comment:
I know this is something that should have been asked in the earlier commit, but is there a reason we're not just doing `type Ref struct { sync.RWMutex }` and then skipping the four pass-thru methods?